app/models/cart.rb

```diff
@@ -23,12 +23,39 @@ def items
   end
 
   def subtotal(item)
-    item.price * @contents[item.id.to_s]
+    if highest_discount(item) != nil
+      discounted(item)
+    else
+      item.price * @contents[item.id.to_s]
+    end
+  end
+
+  def highest_discount(item)
+    ordered_discounts = item.discounts.order(min_quantity: :desc)
+    new_discount = nil
+    ordered_discounts.each do |discount|
+      if @contents[item.id.to_s] >= discount.min_quantity
+        new_discount = discount
+        break
+      end
+    end
+    new_discount
+  end
+
+  def discounted(item)
+    price = item.price * @contents[item.id.to_s]
+    to_percentage = (highest_discount(item).percent_off) / 100.to_f
+    price - (price * to_percentage)
   end
 
   def total
     @contents.sum do |item_id, quantity|
-      Item.find(item_id).price * quantity
+      item = Item.find(item_id)
+      if highest_discount(item) != nil
+        discounted(item)
+      else
+        item.price * quantity
+      end
```

Comment:
Add conditional that will add the items discounted subtotal if it has a discount.